### ISSUE TITLE
[Hotreload] Hotreload breaks shops

### ISSUE THREAD

**Issue by TimGoll** [TTT-2/TTT2#1091]
Shops with standard items (not with manually selected items) are broken after hotreload. This is because the functions has to be called after a hotreload: https://github.com/TTT-2/TTT2/blob/master/gamemodes/terrortown/gamemode/client/cl_equip.lua#L1117-L1130

**Comment by TimGoll** [comment]
It seems to be more tricky then expected. Things I learned so far:

- Recaching the materials on hotreload doesn't solve the issue. For items it works kind of (even though they are not buyable then), but weapons are stil not there
- `.material` of all weapons is `nil`
- `AddEquipmentKeyValues` is not called on hotreload, it sets up the equipment

**Comment by ZenBre4ker** [comment]
I tested this a lot as well and I think there was a problem in the TTT2RegisterSWEP part.

**Comment by TimGoll** [comment]
> I tested this a lot as well and I think there was a problem in the TTT2RegisterSWEP part.

yes, but it is hard to debug because all these functions seem to be called all over the place

**Comment by ZenBre4ker** [comment]
This here fixed the hotreload for weapon stats I remember:
https://github.com/TTT-2/TTT2/commit/b72ae87d6b644280ec73d1f53506b9a5182da65a
